### PR TITLE
Add STATIC_COMPILE flag, make if format consistent

### PR DESCRIPTION
**Pull request by simonsan**
+ integrates the changes for `find_package` of #4

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -12,47 +13,48 @@ project(genieutils)
...
 
-if (GUTILS_TEST)
+if(GUTILS_TEST)
   find_package(Boost 1.55 COMPONENTS unit_test_framework program_options REQUIRED)
   find_package(SFML COMPONENTS SYSTEM WINDOW GRAPHICS REQUIRED)
-endif (GUTILS_TEST)
+endif(GUTILS_TEST)
 
-find_package(Iconv)
+find_package(Iconv REQUIRED)
```
Comment (simonsan) [inline CMakeLists.txt:37]: - deleted the doubled entry from L19 (<https://github.com/Tapsa/genieutils/pull/5/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aL19>)
- added `REQUIRED` from L19 here

File: CMakeLists.txt [hunk 2]
```diff
@@ -12,47 +13,48 @@ project(genieutils)
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR}/cmake/)
 #set(CMAKE_CXX_FLAGS "-std=gnu++0x")
 
-#set(Boost_USE_STATIC_LIBS        ON)
-#set(Boost_USE_STATIC_RUNTIME     ON)
-set(Boost_USE_MULTITHREADED      ON)
+if(STATIC_COMPILE)
+  set(Boost_USE_STATIC_LIBS        ON)
+  #set(Boost_USE_STATIC_RUNTIME     ON)
+endif(STATIC_COMPILE)
```
Comment (simonsan) [inline CMakeLists.txt:19]: Wrapped this to use `-DSTATIC_COMPILE` and no need for `sed`-scripts or patching this repository anymore in `auto-mods`

File: CMakeLists.txt [hunk 3]
```diff
@@ -155,12 +157,18 @@ set(BINCOMP_SRC src/tools/bincompare/bincomp.cpp
...
 #------------------------------------------------------------------------------#
 
-add_library(${Genieutils_LIBRARY} SHARED ${FILE_SRC} ${LANG_SRC} ${DAT_SRC} 
-                                  ${RESOURCE_SRC} ${UTIL_SRC} ${SCRIPT_SRC} )
-target_link_libraries(${Genieutils_LIBRARY} ${ZLIB_LIBRARIES} ${Boost_LIBRARIES} ${ICONV_LIBRARIES})
+if(STATIC_COMPILE)
+  add_library(${Genieutils_LIBRARY} STATIC ${FILE_SRC} ${LANG_SRC} ${DAT_SRC} 
+                                    ${RESOURCE_SRC} ${UTIL_SRC} ${SCRIPT_SRC} )
+  target_link_libraries(${Genieutils_LIBRARY} ${Boost_LIBRARIES} ${ZLIB_LIBRARIES} ${ICONV_LIBRARIES})
+else()
```
Comment (simonsan) [inline CMakeLists.txt:167]: Wrapped that as well for use as `-DSTATIC_COMPILE`, changed order of link libraries as we had problems with it during static linking.

**Comment by Tapsa** [comment]
Thanks for your contribution. I’ll get back on this hopefully soon™

Lähettäjä: simonsan<mailto:notifications@github.com>
Lähetetty: maanantai 1. helmikuuta 2021 13.52
Vastaanottaja: Tapsa/genieutils<mailto:genieutils@noreply.github.com>
Kopio: Mikko P<mailto:mikonposti@live.fi>; Mention<mailto:mention@noreply.github.com>
Aihe: Re: [Tapsa/genieutils] Add STATIC_COMPILE flag, make if format consistent (#5)


r? @Tapsa<https://github.com/Tapsa>

**Comment by Tapsa** [comment]
Thanks for your contribution. I’ll get back on this hopefully soon™

File: CMakeLists.txt (hunk 1 above)
Comment (Tapsa) [reply CMakeLists.txt:37]: Iconv shouldn't be required though. If not found, the conversion code should be disabled.

Comment (simonsan) [reply CMakeLists.txt:37]:
```suggestion
find_package(Iconv)
```

Comment (simonsan) [reply CMakeLists.txt:37]: I added a suggestion for easy committing to the PR.

Comment (Tapsa) [reply CMakeLists.txt:37]: It's all right. Looks like I had it already there, though I remember that it used to be optional somewhere.

**Pull request merged**